### DISCUSSION TITLE
What's the equivalent of get_extra_info("peername"), but for the server side? Tried get_extra_info("sockname"), but no luck.

### DISCUSSION THREAD

**Discussion by DavesCodeMusings** [micropython/micropython#13010] [Using MicroPython] [answered]
I've got some code using `asyncio.start_server()` and I'm trying to determine the port for the new connection on the server side. I have no problem getting info for the client side with `get_extra_info("peername")`, but I cannot figure out how to do it for the server side.

The CPython way seems to be by using `data_writer.get_extra_info("sockname")`, but rather than getting the server side info, I get an exception _KeyError: sockname_. I tried `data_reader.get_extra_info("sockname")` as well, but got the same exception.

What I'm trying to do is figure out what TCP port the client connected to on the server side from within my `on_connect()` function. The code below shows my attempt to find that with `get_extra_info("sockname")`.

Any suggestions on how I can get the server side TCP port number?

Thanks much!

```
    async def on_data_connect(self, data_reader, data_writer):
        """
        Handler for PASV data connections. Remember the streams for later commands.
        """
        client_ip = data_writer.get_extra_info("peername")[0]
        server_side = data_writer.get_extra_info("sockname")
        if self.debug:
            print("Data connection from:", client_ip)
            print("Server side:", server_side)
```

**Comment by glenn20** [comment, marked as answer]
It would seem that micropython does not support the `socket.getsockname()` function (though lwip supports it) and so the `Stream` class can't populate the `"sockname" property of the streamreader.

**Comment by shawwwn** [comment]
Are there plans to add `.getsockname()` and `.getpeername()` to `socket` module? Thanks!

**Reply by sophiedegran** [reply]
On micropython 1.22.2 with asyncio v3
peername is on the StreamReader not the StreamWriter (as it is on python)
so it will be: 
client_ip = data_reader.get_extra_info ('peername')
